model/godleyIcon.h

```diff
@@ -33,15 +33,70 @@
 namespace minsky
 {
   class GodleyTableEditor;
+  
+  // Godley variables broken out in a separate structure, as copying is non-default
+  class GodleyVars: public ItemT<GodleyVars>
+  {
+  public:	    
+    typedef std::vector<VariablePtr> Variables;
+    const Variables& flowVars() const {return m_flowVars;}
+    const Variables& stockVars() const {return m_stockVars;}    	  
+	  
+    GodleyVars() {}
+    virtual ~GodleyVars() {}
+    // copy operations not deleted to allow ItemT<Group> to compile
+    GodleyVars(const GodleyVars& x) {};
+    GodleyVars& operator=(const GodleyVars&) {return *this;}
+    classdesc::Exclude<std::weak_ptr<GodleyIcon>> self; ///< weak ref to this    
+    
+    void clear() {
+      m_flowVars.clear();
+      m_stockVars.clear();
+    }  
+    
+    bool variableDisplay=true;
+    void toggleVariableDisplay() {variableDisplay=!variableDisplay;}   
+    
+    void removeControlledItems() const override;    
+    
+    /// flows, along with multipliers, appearing in \a col
+    std::map<string,double> flowSignature(int col) const;    
+    
+    GodleyTable table;
+    
+    /// updates the variable lists with the Godley table
+    void update(); 
```

Comment:
Make this a virtual function, as it needs to be implemented in GodleyIcon, but callable in this class to implement the copy operations.